In [1]:
# Utils
# =============================================================================
from utils.paths import (
    ORDERS_FILE, 
    ORDER_ITEMS_FILE, 
    PAYMENTS_FILE, 
    REVIEWS_FILE, 
    CUSTOMERS_FILE, 
    PRODUCTS_FILE, 
    SELLERS_FILE, 
    GEOLOCATION_FILE, 
    CATEGORY_TRANSLATION_FILE
)

# DATA
# =============================================================================
import pandas as pd

# EDA
# =============================================================================
from data_assessmen.data_profiling import (
    evaluate_completeness,
    evaluate_uniqueness,
    evaluate_distribution)

# **Data**
---
**`OLIST_FILES`**

A dictionary that maps each table's short internal name (used throughout the code) to its actual CSV filename as distributed by Kaggle.

Keeping this mapping in one dictionary avoids hardcoding filenames in multiple places, and lets the loader iterate over all tables with a single loop instead of repeating near-identical code nine times.

| Key | File | Description |
|---|---|---|
| `orders` | `olist_orders_dataset.csv` | Core orders table; source of order status and timestamps |
| `order_items` | `olist_order_items_dataset.csv` | Line items linked to each order, at product level |
| `payments` | `olist_order_payments_dataset.csv` | Payment records linked to each order |
| `reviews` | `olist_order_reviews_dataset.csv` | Customer reviews linked to each order |
| `customers` | `olist_customers_dataset.csv` | Customer records linked to each order |
| `products` | `olist_products_dataset.csv` | Product catalog linked to order items |
| `sellers` | `olist_sellers_dataset.csv` | Seller records linked to order items |
| `geolocation` | `olist_geolocation_dataset.csv` | Reference table of zip-code coordinates |
| `category_translation` | `product_category_name_translation.csv` | Reference table translating product categories |

In [2]:
orders = pd.read_csv(ORDERS_FILE)
orders_items = pd.read_csv(ORDER_ITEMS_FILE)
payments_file = pd.read_csv(PAYMENTS_FILE)
reviews_file = pd.read_csv(REVIEWS_FILE)
customers_file = pd.read_csv(CUSTOMERS_FILE)
products_file = pd.read_csv(PRODUCTS_FILE)
sellers_file = pd.read_csv(SELLERS_FILE)
geolocation_file = pd.read_csv(GEOLOCATION_FILE)
category_translation_file = pd.read_csv(CATEGORY_TRANSLATION_FILE)

# **Completeness**
---

In [3]:
evaluate_completeness(orders)

,Dimension,Column,Record Count,Value Count,Null Count,Completeness %
0,Completeness,order_id,99441,99441,0,100.00
1,Completeness,customer_id,99441,99441,0,100.00
2,Completeness,order_status,99441,99441,0,100.00
3,Completeness,order_purchase_timestamp,99441,99441,0,100.00
4,Completeness,order_approved_at,99441,99281,160,99.84
5,Completeness,order_delivered_carrier_date,99441,97658,1783,98.21
6,Completeness,order_delivered_customer_date,99441,96476,2965,97.02
7,Completeness,order_estimated_delivery_date,99441,99441,0,100.00


`orders` is fully complete for identifiers, status, and the purchase/estimated-delivery timestamps. The remaining gaps sit in the fulfillment timeline `order_approved_at` (0.16% missing), `order_delivered_carrier_date` (1.79%), and `order_delivered_customer_date` (2.98%) consistent with orders that were canceled or never reached final delivery.

In [4]:
evaluate_completeness(orders_items)

,Dimension,Column,Record Count,Value Count,Null Count,Completeness %
0,Completeness,order_id,112650,112650,0,100.0
1,Completeness,order_item_id,112650,112650,0,100.0
2,Completeness,product_id,112650,112650,0,100.0
3,Completeness,seller_id,112650,112650,0,100.0
4,Completeness,shipping_limit_date,112650,112650,0,100.0
5,Completeness,price,112650,112650,0,100.0
6,Completeness,freight_value,112650,112650,0,100.0


`order_items` has no missing values in any column, meaning every line item has a valid order, product, seller, shipping date, price, and freight value.

In [5]:
evaluate_completeness(payments_file)

,Dimension,Column,Record Count,Value Count,Null Count,Completeness %
0,Completeness,order_id,103886,103886,0,100.0
1,Completeness,payment_sequential,103886,103886,0,100.0
2,Completeness,payment_type,103886,103886,0,100.0
3,Completeness,payment_installments,103886,103886,0,100.0
4,Completeness,payment_value,103886,103886,0,100.0


`payments` is fully complete across all columns, so every payment record has a valid order, type, installment count, and value.

In [6]:
evaluate_completeness(reviews_file)

,Dimension,Column,Record Count,Value Count,Null Count,Completeness %
0,Completeness,review_id,99224,99224,0,100.00
1,Completeness,order_id,99224,99224,0,100.00
2,Completeness,review_score,99224,99224,0,100.00
3,Completeness,review_comment_title,99224,11568,87656,11.66
4,Completeness,review_comment_message,99224,40977,58247,41.30
5,Completeness,review_creation_date,99224,99224,0,100.00
6,Completeness,review_answer_timestamp,99224,99224,0,100.00


`reviews` is fully complete for structural fields (ids, score, dates), but the free-text fields are largely empty: `review_comment_title` is missing 88.34% of the time and `review_comment_message` 58.70%. This reflects that most customers leave a numeric score without writing a comment, so any text-based analysis on this table will only cover a minority of reviews.

In [7]:
evaluate_completeness(customers_file)

,Dimension,Column,Record Count,Value Count,Null Count,Completeness %
0,Completeness,customer_id,99441,99441,0,100.0
1,Completeness,customer_unique_id,99441,99441,0,100.0
2,Completeness,customer_zip_code_prefix,99441,99441,0,100.0
3,Completeness,customer_city,99441,99441,0,100.0
4,Completeness,customer_state,99441,99441,0,100.0


`customers` has no missing values, so every customer record carries a complete id, zip code, city, and state.

In [8]:
evaluate_completeness(products_file)

,Dimension,Column,Record Count,Value Count,Null Count,Completeness %
0,Completeness,product_id,32951,32951,0,100.00
1,Completeness,product_category_name,32951,32341,610,98.15
2,Completeness,product_name_lenght,32951,32341,610,98.15
3,Completeness,product_description_lenght,32951,32341,610,98.15
4,Completeness,product_photos_qty,32951,32341,610,98.15
5,Completeness,product_weight_g,32951,32949,2,99.99
6,Completeness,product_length_cm,32951,32949,2,99.99
7,Completeness,product_height_cm,32951,32949,2,99.99
8,Completeness,product_width_cm,32951,32949,2,99.99


`products` has two small clusters of missing data: the descriptive attributes (`product_category_name`, name/description length, photos qty) are missing in 1.85% of rows likely products cataloged without full metadata while the physical dimensions (weight, length, height, width) are missing in only 2 rows, a negligible edge case.

In [9]:
evaluate_completeness(sellers_file)

,Dimension,Column,Record Count,Value Count,Null Count,Completeness %
0,Completeness,seller_id,3095,3095,0,100.0
1,Completeness,seller_zip_code_prefix,3095,3095,0,100.0
2,Completeness,seller_city,3095,3095,0,100.0
3,Completeness,seller_state,3095,3095,0,100.0


`sellers` is fully complete, with every seller record containing a valid id, zip code, city, and state.

In [10]:
evaluate_completeness(geolocation_file)

,Dimension,Column,Record Count,Value Count,Null Count,Completeness %
0,Completeness,geolocation_zip_code_prefix,1000163,1000163,0,100.0
1,Completeness,geolocation_lat,1000163,1000163,0,100.0
2,Completeness,geolocation_lng,1000163,1000163,0,100.0
3,Completeness,geolocation_city,1000163,1000163,0,100.0
4,Completeness,geolocation_state,1000163,1000163,0,100.0


`geolocation` is fully complete across its ~1M rows, with no missing zip codes, coordinates, city, or state values.

In [11]:
evaluate_completeness(category_translation_file)

,Dimension,Column,Record Count,Value Count,Null Count,Completeness %
0,Completeness,product_category_name,71,71,0,100.0
1,Completeness,product_category_name_english,71,71,0,100.0


`category_translation` is fully complete, with all 71 categories mapped to their English equivalent.

# **Uniqueness**
---

In [12]:
evaluate_uniqueness(orders)

,Dimension,Column,Record Count,Unique Count,Duplicate Count,Uniqueness %
0,Uniqueness,order_id,99441,99441,0,100.00
1,Uniqueness,customer_id,99441,99441,0,100.00
2,Uniqueness,order_status,99441,8,99433,0.01
3,Uniqueness,order_purchase_timestamp,99441,98875,566,99.43
4,Uniqueness,order_approved_at,99441,90734,8707,91.24
5,Uniqueness,order_delivered_carrier_date,99441,81019,18422,81.47
6,Uniqueness,order_delivered_customer_date,99441,95665,3776,96.20
7,Uniqueness,order_estimated_delivery_date,99441,459,98982,0.46


`order_id` and `customer_id` are fully unique, confirming `orders` has one row per order and each order has its own customer record. Low-cardinality columns like `order_status` (8 values) and `order_estimated_delivery_date` (459 distinct dates) show expected repetition rather than a data quality issue.

In [13]:
evaluate_uniqueness(orders_items)

,Dimension,Column,Record Count,Unique Count,Duplicate Count,Uniqueness %
0,Uniqueness,order_id,112650,98666,13984,87.59
1,Uniqueness,order_item_id,112650,21,112629,0.02
2,Uniqueness,product_id,112650,32951,79699,29.25
3,Uniqueness,seller_id,112650,3095,109555,2.75
4,Uniqueness,shipping_limit_date,112650,93318,19332,82.84
5,Uniqueness,price,112650,5968,106682,5.30
6,Uniqueness,freight_value,112650,6999,105651,6.21


`order_id` repeats across 87.59% of rows because orders can contain multiple items, which is the expected one-to-many relationship with `orders`. `product_id` and `seller_id` also repeat naturally, as the same product or seller appears in many line items.

In [14]:
evaluate_uniqueness(payments_file)

,Dimension,Column,Record Count,Unique Count,Duplicate Count,Uniqueness %
0,Uniqueness,order_id,103886,99440,4446,95.72
1,Uniqueness,payment_sequential,103886,29,103857,0.03
2,Uniqueness,payment_type,103886,5,103881,0.00
3,Uniqueness,payment_installments,103886,24,103862,0.02
4,Uniqueness,payment_value,103886,29077,74809,27.99


`order_id` is duplicated in 4.28% of rows, consistent with orders that were split across multiple payment methods or installments (`payment_sequential` up to 29). `payment_type` has only 5 distinct values, as expected for a categorical field.

In [15]:
evaluate_uniqueness(reviews_file)

,Dimension,Column,Record Count,Unique Count,Duplicate Count,Uniqueness %
0,Uniqueness,review_id,99224,98410,814,99.18
1,Uniqueness,order_id,99224,98673,551,99.44
2,Uniqueness,review_score,99224,5,99219,0.01
3,Uniqueness,review_comment_title,99224,4528,94696,4.56
4,Uniqueness,review_comment_message,99224,36160,63064,36.44
5,Uniqueness,review_creation_date,99224,636,98588,0.64
6,Uniqueness,review_answer_timestamp,99224,98248,976,99.02


`review_id` is duplicated in 0.82% of rows and `order_id` in 0.56%, suggesting a small number of orders received more than one review. `review_score` behaves as expected with only 5 distinct values (1–5).

In [16]:
evaluate_uniqueness(customers_file)

,Dimension,Column,Record Count,Unique Count,Duplicate Count,Uniqueness %
0,Uniqueness,customer_id,99441,99441,0,100.00
1,Uniqueness,customer_unique_id,99441,96096,3345,96.64
2,Uniqueness,customer_zip_code_prefix,99441,14994,84447,15.08
3,Uniqueness,customer_city,99441,4119,95322,4.14
4,Uniqueness,customer_state,99441,27,99414,0.03


`customer_id` is fully unique (one row per order-level customer record), while `customer_unique_id` repeats in 3.36% of rows  this is the key signal that the same physical customer placed multiple orders under different `customer_id` values, which matters for any repeat-purchase analysis.

In [17]:
evaluate_uniqueness(products_file)

,Dimension,Column,Record Count,Unique Count,Duplicate Count,Uniqueness %
0,Uniqueness,product_id,32951,32951,0,100.00
1,Uniqueness,product_category_name,32951,74,32877,0.22
2,Uniqueness,product_name_lenght,32951,67,32884,0.20
3,Uniqueness,product_description_lenght,32951,2961,29990,8.99
4,Uniqueness,product_photos_qty,32951,20,32931,0.06
5,Uniqueness,product_weight_g,32951,2205,30746,6.69
6,Uniqueness,product_length_cm,32951,100,32851,0.30
7,Uniqueness,product_height_cm,32951,103,32848,0.31
8,Uniqueness,product_width_cm,32951,96,32855,0.29


`product_id` is fully unique, confirming one row per product. The remaining columns are low-cardinality descriptive attributes (74 categories, dimension/weight buckets), so their repetition is expected rather than a duplication problem.

In [18]:
evaluate_uniqueness(sellers_file)

,Dimension,Column,Record Count,Unique Count,Duplicate Count,Uniqueness %
0,Uniqueness,seller_id,3095,3095,0,100.00
1,Uniqueness,seller_zip_code_prefix,3095,2246,849,72.57
2,Uniqueness,seller_city,3095,611,2484,19.74
3,Uniqueness,seller_state,3095,23,3072,0.74


`seller_id` is fully unique. `seller_zip_code_prefix` repeats in 27.43% of rows, and `seller_city`/`seller_state` repeat even more, since many sellers share the same location — an expected geographic concentration.

In [19]:
evaluate_uniqueness(geolocation_file)

,Dimension,Column,Record Count,Unique Count,Duplicate Count,Uniqueness %
0,Uniqueness,geolocation_zip_code_prefix,1000163,19015,981148,1.90
1,Uniqueness,geolocation_lat,1000163,717360,282803,71.72
2,Uniqueness,geolocation_lng,1000163,717613,282550,71.75
3,Uniqueness,geolocation_city,1000163,8011,992152,0.80
4,Uniqueness,geolocation_state,1000163,27,1000136,0.00


No column in `geolocation` is unique, and `geolocation_zip_code_prefix` repeats in 98.10% of rows — this table stores many lat/lng points per zip code (a reference/lookup table), so heavy repetition is by design, not a defect.

In [20]:
evaluate_uniqueness(category_translation_file)

,Dimension,Column,Record Count,Unique Count,Duplicate Count,Uniqueness %
0,Uniqueness,product_category_name,71,71,0,100.0
1,Uniqueness,product_category_name_english,71,71,0,100.0


Both columns are fully unique, confirming a clean 1-to-1 mapping between each Portuguese category name and its English translation, with no duplicate entries.

# **Distribution**
---

In [21]:
evaluate_distribution(orders)

,Dimension,Column,Data Type,Min,Max,Mean,Median,Std,Unique,Top Value,Top %,Days,Months,Years
0,Distribution,order_id,Categorical,None,None,None,None,None,99441,66dea50a8b16d9b4dee7af250b4be1a5,0.00,None,None,None
1,Distribution,customer_id,Categorical,None,None,None,None,None,99441,edb027a75a1449115f6b43211ae02a24,0.00,None,None,None
2,Distribution,order_status,Categorical,None,None,None,None,None,8,delivered,97.02,None,None,None
3,Distribution,order_purchase_timestamp,Categorical,None,None,None,None,None,98875,2018-02-19 15:37:47,0.00,None,None,None
4,Distribution,order_approved_at,Categorical,None,None,None,None,None,90733,2018-02-27 04:31:10,0.01,None,None,None
5,Distribution,order_delivered_carrier_date,Categorical,None,None,None,None,None,81018,2018-05-09 15:48:00,0.05,None,None,None
6,Distribution,order_delivered_customer_date,Categorical,None,None,None,None,None,95664,2016-10-27 17:32:07,0.00,None,None,None
7,Distribution,order_estimated_delivery_date,Categorical,None,None,None,None,None,459,2017-12-20 00:00:00,0.52,None,None,None


All columns in `orders` are treated as categorical (ids and timestamps), so no numeric stats apply. `order_status` is dominated by `delivered` (97.02%), meaning cancellations and other non-delivered states are a small minority of the dataset important to keep in mind when using this table for delivery-time analysis.

In [22]:
evaluate_distribution(orders_items)

,Dimension,Column,Data Type,Min,Max,Mean,Median,Std,Unique,Top Value,Top %,Days,Months,Years
0,Distribution,order_id,Categorical,NaN,NaN,NaN,NaN,NaN,98666.0,8272b63d03f5f79c56e9e4120aec44ef,0.02,None,None,None
1,Distribution,order_item_id,Numeric,1.00,21.00,1.20,1.00,0.71,NaN,None,NaN,None,None,None
2,Distribution,product_id,Categorical,NaN,NaN,NaN,NaN,NaN,32951.0,aca2eb7d00ea1a7b8ebd4e68314663af,0.47,None,None,None
3,Distribution,seller_id,Categorical,NaN,NaN,NaN,NaN,NaN,3095.0,6560211a19b47992c3666cc44a7e94c0,1.80,None,None,None
4,Distribution,shipping_limit_date,Categorical,NaN,NaN,NaN,NaN,NaN,93318.0,2018-03-01 02:50:48,0.02,None,None,None
5,Distribution,price,Numeric,0.85,6735.00,120.65,74.99,183.63,NaN,None,NaN,None,None,None
6,Distribution,freight_value,Numeric,0.00,409.68,19.99,16.26,15.81,NaN,None,NaN,None,None,None


`price` ranges from 0.85 to 6,735.00 with a mean of 120.65 well above the median of 74.99, indicating a right-skewed distribution with high-value outliers. `freight_value` shows the same pattern (mean 19.99 vs. median 16.26). Most orders (mean `order_item_id` of 1.20) contain very few items.

In [23]:
evaluate_distribution(payments_file)

,Dimension,Column,Data Type,Min,Max,Mean,Median,Std,Unique,Top Value,Top %,Days,Months,Years
0,Distribution,order_id,Categorical,NaN,NaN,NaN,NaN,NaN,99440.0,fa65dad1b0e818e3ccc5cb0e39231352,0.03,None,None,None
1,Distribution,payment_sequential,Numeric,1.0,29.00,1.09,1.0,0.71,NaN,None,NaN,None,None,None
2,Distribution,payment_type,Categorical,NaN,NaN,NaN,NaN,NaN,5.0,credit_card,73.92,None,None,None
3,Distribution,payment_installments,Numeric,0.0,24.00,2.85,1.0,2.69,NaN,None,NaN,None,None,None
4,Distribution,payment_value,Numeric,0.0,13664.08,154.10,100.0,217.49,NaN,None,NaN,None,None,None


`credit_card` dominates `payment_type` at 73.92%, making it by far the preferred payment method. `payment_value` is strongly right-skewed (mean 154.10 vs. median 100.00, max 13,664.08), and `payment_installments` averages under 3, though it can go up to 24.

In [24]:
evaluate_distribution(reviews_file)

,Dimension,Column,Data Type,Min,Max,Mean,Median,Std,Unique,Top Value,Top %,Days,Months,Years
0,Distribution,review_id,Categorical,NaN,NaN,NaN,NaN,NaN,98410.0,4548534449b1f572e357211b90724f1b,0.00,None,None,None
1,Distribution,order_id,Categorical,NaN,NaN,NaN,NaN,NaN,98673.0,8e17072ec97ce29f0e1f111e598b0c85,0.00,None,None,None
2,Distribution,review_score,Numeric,1.0,5.0,4.09,5.0,1.35,NaN,None,NaN,None,None,None
3,Distribution,review_comment_title,Categorical,NaN,NaN,NaN,NaN,NaN,4527.0,Recomendo,3.66,None,None,None
4,Distribution,review_comment_message,Categorical,NaN,NaN,NaN,NaN,NaN,36159.0,Muito bom,0.56,None,None,None
5,Distribution,review_creation_date,Categorical,NaN,NaN,NaN,NaN,NaN,636.0,2017-12-19 00:00:00,0.47,None,None,None
6,Distribution,review_answer_timestamp,Categorical,NaN,NaN,NaN,NaN,NaN,98248.0,2017-06-15 23:21:05,0.00,None,None,None


`review_score` has a mean of 4.09 and median of 5.0, showing customers skew heavily toward positive reviews. Combined with the low completeness of comment fields seen earlier, this suggests most reviews are quick, positive ratings without detailed written feedback.

In [25]:
evaluate_distribution(customers_file)

,Dimension,Column,Data Type,Min,Max,Mean,Median,Std,Unique,Top Value,Top %,Days,Months,Years
0,Distribution,customer_id,Categorical,NaN,NaN,NaN,NaN,NaN,99441.0,274fa6071e5e17fe303b9748641082c8,0.00,None,None,None
1,Distribution,customer_unique_id,Categorical,NaN,NaN,NaN,NaN,NaN,96096.0,8d50f5eadf50201ccdcedfb9e2ac8455,0.02,None,None,None
2,Distribution,customer_zip_code_prefix,Numeric,1003.0,99990.0,35137.47,24416.0,29797.94,NaN,None,NaN,None,None,None
3,Distribution,customer_city,Categorical,NaN,NaN,NaN,NaN,NaN,4119.0,sao paulo,15.63,None,None,None
4,Distribution,customer_state,Categorical,NaN,NaN,NaN,NaN,NaN,27.0,SP,41.98,None,None,None


`customer_state` is heavily concentrated in `SP` (41.98%), and `sao paulo` is also the top city (15.63%), confirming the well-known concentration of Brazilian e-commerce activity in São Paulo state. `customer_zip_code_prefix` spans the full national range (1,003–99,990).

In [26]:
evaluate_distribution(products_file)

,Dimension,Column,Data Type,Min,Max,Mean,Median,Std,Unique,Top Value,Top %,Days,Months,Years
0,Distribution,product_id,Categorical,NaN,NaN,NaN,NaN,NaN,32951.0,106392145fca363410d287a815be6de4,0.00,None,None,None
1,Distribution,product_category_name,Categorical,NaN,NaN,NaN,NaN,NaN,73.0,cama_mesa_banho,9.37,None,None,None
2,Distribution,product_name_lenght,Numeric,5.0,76.0,48.48,51.0,10.25,NaN,None,NaN,None,None,None
3,Distribution,product_description_lenght,Numeric,4.0,3992.0,771.50,595.0,635.12,NaN,None,NaN,None,None,None
4,Distribution,product_photos_qty,Numeric,1.0,20.0,2.19,1.0,1.74,NaN,None,NaN,None,None,None
5,Distribution,product_weight_g,Numeric,0.0,40425.0,2276.47,700.0,4282.04,NaN,None,NaN,None,None,None
6,Distribution,product_length_cm,Numeric,7.0,105.0,30.82,25.0,16.91,NaN,None,NaN,None,None,None
7,Distribution,product_height_cm,Numeric,2.0,105.0,16.94,13.0,13.64,NaN,None,NaN,None,None,None
8,Distribution,product_width_cm,Numeric,6.0,118.0,23.20,20.0,12.08,NaN,None,NaN,None,None,None


`cama_mesa_banho` (bed, bath & table) is the top category at 9.37% of products. `product_weight_g` is extremely right-skewed (mean 2,276.47 vs. median 700, max 40,425), meaning a few heavy items pull the average well above typical products. Physical dimensions (length, height, width) show more moderate spread.

In [27]:
evaluate_distribution(sellers_file)

,Dimension,Column,Data Type,Min,Max,Mean,Median,Std,Unique,Top Value,Top %,Days,Months,Years
0,Distribution,seller_id,Categorical,NaN,NaN,NaN,NaN,NaN,3095.0,9e25199f6ef7e7c347120ff175652c3b,0.03,None,None,None
1,Distribution,seller_zip_code_prefix,Numeric,1001.0,99730.0,32291.06,14940.0,32713.45,NaN,None,NaN,None,None,None
2,Distribution,seller_city,Categorical,NaN,NaN,NaN,NaN,NaN,611.0,sao paulo,22.42,None,None,None
3,Distribution,seller_state,Categorical,NaN,NaN,NaN,NaN,NaN,23.0,SP,59.74,None,None,None


`seller_state` is even more concentrated than customers, with `SP` accounting for 59.74% of sellers, and `sao paulo` as the top city (22.42%)  sellers are more geographically concentrated than buyers, likely reflecting where sourcing/logistics hubs are located.

In [28]:
evaluate_distribution(geolocation_file)

,Dimension,Column,Data Type,Min,Max,Mean,Median,Std,Unique,Top Value,Top %,Days,Months,Years
0,Distribution,geolocation_zip_code_prefix,Numeric,1001.00,99990.00,36574.17,26530.00,30549.34,NaN,None,NaN,None,None,None
1,Distribution,geolocation_lat,Numeric,-36.61,45.07,-21.18,-22.92,5.72,NaN,None,NaN,None,None,None
2,Distribution,geolocation_lng,Numeric,-101.47,121.11,-46.39,-46.64,4.27,NaN,None,NaN,None,None,None
3,Distribution,geolocation_city,Categorical,NaN,NaN,NaN,NaN,NaN,8011.0,sao paulo,13.58,None,None,None
4,Distribution,geolocation_state,Categorical,NaN,NaN,NaN,NaN,NaN,27.0,SP,40.42,None,None,None


Latitude and longitude ranges (roughly -36 to 45, and -101 to 121) extend beyond Brazil's actual borders, suggesting a small number of coordinate outliers or data errors that should be filtered before using this table for geospatial analysis. `SP` again leads as the top state (40.42%), consistent with the other tables.

In [29]:
evaluate_distribution(category_translation_file)

,Dimension,Column,Data Type,Min,Max,Mean,Median,Std,Unique,Top Value,Top %,Days,Months,Years
0,Distribution,product_category_name,Categorical,None,None,None,None,None,71,beleza_saude,1.41,None,None,None
1,Distribution,product_category_name_english,Categorical,None,None,None,None,None,71,health_beauty,1.41,None,None,None


With 71 categories evenly represented (top category at only 1.41%), `category_translation` is a flat reference table with no dominant class, as expected for a lookup/mapping table.